text/0000-deployment-language.md

```diff
@@ -0,0 +1,561 @@
+
+```
+"/source/system::stdin" -> "/pipeline/system::passthrough" -> "/sink/system::stderr";
+```
+
+This is equivalent to the following, when adding explicit parens to show precedence:
+
+```
+("/source/system::stdin/out" -> "/pipeline/system::passthrough/in") -> "/sink/system::stderr/in";
+```
+
+Which resolves via desugaring:
+
+```
+connect "/source/system::stdin/out" to "/pipeline/system::passthrough/in";
+connect "/pipeline/system::passthrough/out" to "/sink/system::stderr/in";
+```
+
+`Arrow` statements also support tuples of Tremor URLs or tuples of other `Arrow` statements. These describe branching and joining
+at the troy level.
+
+```
+"/source/system::stdin/out" -> ("/pipeline/system::passthrough", "/pipeline/my_pipe" ) -> "/sink/system::stderr/in";
+```
+
+This desugars to:
+
+```
+"/source/system::stdin/out" -> "/pipeline/system::passthrough" -> "/sink/system::stderr/in";
+"/source/system::stdin/out" -> "/pipeline/my_pipe" -> "/sink/system::stderr/in";
+```
+
+If we have multiple tuples within a statement, we create statements for each combination of them. Example:
+
+```
+# full sugar
+("/source/my_file", "/source/my_other_file") -> ("/pipeline/pipe1", "/pipeline/pipe2") -> "/sink/system::stderr";
+
+# desugars to:
+
+"/source/my_file" -> "/pipeline/pipe1" -> "/sink/system::stderr";
+"/source/my_file" -> "/pipeline/pipe2" -> "/sink/system::stderr";
+"/source/my_other_file" -> "/pipeline/pipe1" -> "/sink/system::stderr";
+"/source/my_other_file" -> "/pipeline/pipe2" -> "/sink/system::stderr";
+```
+
+The immediate win in terseness is obvious, we hope.
+
+It will be very interesting to explore how to expose the same sugar to trickle select queries. This is a future possibility.
+
+## Instantiating artefacts
+
+Every artefact that was defined can be created, optionally passing arguments:
+
+
+EBNF:
+
+```
+CREATE          = "create" "instance" instance_id "from" artefact_type "/" artefact_id
+                  [ "with" assignment_list ]
+                  "end"
+assignment_list = assignment [ "," assignment_list ]
+assignment      = let_path "=" expr
+let_path        = id [ "." id ]
+```
+
+The result is a running instance that is registered via its `instance-id` within the registry and thus globally resolvable.
+
+**Examples:**
+
+```
+create instance "01" from pipeline/my_pipeline
+with
+    window_size = 15
+end;
+```
+
+It might be interesting to use a tremor URL instead of ids as artefact_type and artefact id. This is an open question.
+
+### Instantiating Flows
+
+Flows are an exception as their creation will possibly create other artefact instances.
+
+## Top-Level Connect statements
+
+Connect statements have been introduced as part of `Flow` definitions. It will greatly simplify setting up tremor installations if we could write them at the top level of a troy script.
```

Comment:
```suggestion
Connect statements have been introduced as part of `flow` definitions. It will greatly simplify setting up tremor installations if we could write them at the top level of a troy script.
```